### ISSUE TITLE
Composite geometry cannot refer to objects defined in include file

### ISSUE THREAD

**Issue by ftessier** [nrc-cnrc/EGSnrc#300]
Input files in egs++ can contain the key-value pair `includefile = filename`, which inserts the content of file `filename`, as if the content of the file had been written in the input file. However, composite geometries cannot find geometries defined in the include file. At least the envelope geometry suffers this problem. For example:

**box2.egsinp**
```ruby
    ### box 2
    :start geometry:
        name     = box2
        library  = egs_box
        box size = 1 1 1
        :start media input:
            media = medium2
        :stop media input:
    :stop geometry:
```
**include.egsinp**
```ruby
:start geometry definition:

    ### box 1
    :start geometry:
        name     = box1
        library  = egs_box
        box size =   2 2 2
        :start media input:
            media = medium1
        :stop media input:
    :stop geometry:

    includefile = box2.egsinp

    ### insert box2 in box1
    :start geometry:
        name                 = box2_in_box1
        library              = egs_genvelope
        base geometry        = box1
        inscribed geometries = box2
    :stop geometry:

    simulation geometry = box2_in_box1   # error: no geometry with name box2 defined

:stop geometry definition:
```
Trying to view the geometry **include.egsinp** yields a geometry not found error:
```bash
$ egs_view include.egsinp 
createGeometry(envelope geometry): no geometry with name box2 defined
```
(Reported by Patrick Saull.)

**Comment by randlet** [comment]
FWIW this was my motivation for creating the [egs_glib](https://github.com/randlet/EGSnrc/blob/946b5c0a554ba43d3ba1f536dea0d96327c5d881/HEN_HOUSE/egs%2B%2B/geometry/egs_glib/egs_glib.h#L55) geometry type.

**Comment by ftessier** [comment]
Ah yes! Excellent point; I had forgotten about that!

**Comment by crcrewso** [comment]
This is fantastic for me!!! Consistency without hours of medit here I come!!!

**Comment by ftessier** [comment]
This was resolved in commit 9bdd1b7461cd4b1f4e5df461c54b2c0a159c7b4d.
